giraffa-core/src/main/java/org/apache/giraffa/LeaseManager.java

```diff
@@ -93,4 +107,92 @@ public boolean isLeaseSoftLimitExpired(String holder) {
   public Collection<FileLease> getLeases(String holder) {
     return hlmAdapter.getLeases(holder);
   }
+
+  public synchronized Daemon getMonitorDaemon(NamespaceProcessor namesystem) {
+    if(monitor != null)
+      return monitor;
+
+    Daemon monitorDaemon = new Daemon(new LeaseMonitor(namesystem));
```

Comment:
I don't think we need to create `monitorDaemon` here; we can just update and access the `monitor` field directly. All access to `monitor` is synchronized anyway. Otherwise, I have checked the synchronization logic for this class and it looks good.